File: dehaze-go/model/vo/algorithm.go
package vo

// AlgorithmVO 算法视图对象
type AlgorithmVO struct {
	// 算法ID
	ID int64 `json:"id"`
	// 算法名称
	Name string `json:"name"`
	// 算法类型
	Type string `json:"type"`
	// 算法图片
	Img string `json:"img"`
	// 算法描述
	Description string `json:"description"`
	// 算法路径
	Path string `json:"path"`
	// 算法浮点数
	Flops string `json:"flops"`
	// 算法参数量
	Params string `json:"params"`
	// 导入路径
	ImportPath string `json:"importPath"`
	// 开启关闭状态
	Status int `json:"status"`
	// 算法大小
	Size string `json:"size"`
	// 子算法
	Children []AlgorithmVO `json:"children"`
}

File: dehaze-android/sdk/src/main/java/com/pei/dehaze/sdk/model/algorithm/Algorithm.java
package com.pei.dehaze.sdk.model.algorithm;

import lombok.Data;
import java.util.List;

/**
 * 算法模型类
 */
@Data
public class Algorithm {
    private Long id;
    private Long parentId;
    private String name;
    private String type;
    private String description;
    private String img;
    private String path;
    private String importPath;
    private String params;
    private String flops;
    private AlgorithmStatus status;
    private String size;
    private List<Algorithm> children;
}